test/integration/framework/logger.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"flag"
+	"io"
+	"testing"
+
+	"k8s.io/klog/v2"
+)
+
+// RedirectKlog modifies the global klog logger so that it writes via
+// tb.Log. This only works when different tests run sequentially and causes
+// this file to be logged by testing.TB, so this is inferior compared to
+// per-test output with klog/ktesting. As a stop-gap solution until contextual
```

Comment:
This is being used in this PR to send the output of another binary into the contextual logging and it has very little to do with k/k code not being fully converted to contextual logging (which was my initial guess).

It's unclear to me how this is a stop-gap solution. Do you have an open issue for the final solution?

Reply:
> This is being used in this PR to send the output of another binary into the contextual logging

In practice, that's one line.

> and it has very little to do with k/k code not being fully converted to contextual logging

There are hundreds of lines of output from k/k code right now that get redirected with this because they haven't been converted to contextual logging.

> It's unclear to me how this is a stop-gap solution. Do you have an open issue for the final solution?

True, output redirection of etcd will be needed forever. Let me take out that comment. It'll become less important over time as more code gets converted to contextual logging (tracked by https://github.com/kubernetes/enhancements/issues/3077), but it'll never go away completely.